### ISSUE TITLE
custom-set: split test cases out into their own specs

### ISSUE THREAD

**Issue by ireddick** [exercism/xjavascript#290]
While working on issue #283 we noticed that the specs for custom-set contain multiple test cases and assertions.

This makes it hard for the person doing the exercise to know which test case has actually failed. It also makes it hard for maintainers for similar reasons.

The suggestion is to split each test case out into it's own 'it' and use the canonical test descriptions to guide the new spec names.

For example:

``` javascript
xit('can test for a member', function() {
    var actual = new CustomSet().contains(1);
    expect(actual).toBe(false);

    var actual2 = new CustomSet([1, 2, 3]).contains(1);
    expect(actual2).toBe(true);

    var actual3 = new CustomSet([1, 2, 3]).contains(4);
    expect(actual3).toBe(false);
});
```

might become something like:

``` javascript
xit('nothing is contained in an empty set', function() {
    var actual = new CustomSet().contains(1);
    expect(actual).toBe(false);
});
xit('when the element is in the set', function() {
    var actual = new CustomSet([1, 2, 3]).contains(1);
    expect(actual).toBe(true);
});
xit('when the element is not in the set', function() {
    var actual = new CustomSet([1, 2, 3]).contains(4);
    expect(actual).toBe(false);
});
```

(I think I'd prefer if the descriptions fitted the 'it' syntax more, but I'd need to have a think)

The initial discussion can be found here: https://github.com/exercism/xjavascript/pull/285#issuecomment-228965658

**Comment by ireddick** [comment]
I've had a look around, and it looks like the ruby version is moving to this format:

https://github.com/exercism/xruby/commit/db189a63b05241248f0c8146709cd52dd987bdb7

The scala version is already in this format:

https://github.com/exercism/xscala/blob/master/exercises/custom-set/src/test/scala/CustomSetTest.scala

Both of the above also use the canonical test descriptions for the test names (the ruby version uses a generator to create these).

With this in mind, I think the approach suggested in the initial comment is the way to go. Many of the existing javascript exercises don't attempt to have the 'describe/it' read properly anyway, so it's not a huge issue - we probably want to look into generating the tests anyway.

**Comment by rchavarria** [comment]
Ok, if I understood right, you're not going to modify too much test names from the canonical tests.

About the generator, it would be great to have it. I didn't dig into that, so I'm not sure if there must be only one generator for all exercises, or one generator by exercise. In case of create a generator, I would implement it to follow the `describe`/`it` blocks, because the fit perfectly with canonical test descriptions.

I found this about [exercise generators](https://github.com/exercism/todo/issues/8), it might be a nice point to start with.

**Comment by ireddick** [comment]
Hello - yes, I think following the canonical test descriptions is the way to go.

I'll start by looking into creating an exercise generator.

**Comment by rchavarria** [comment]
Described changes in the issue were added. And the exercise has a generator (included in #292), so that it'll be updated easily if canonical tests change. 

I think this issue is ready to be closed.
